website/content/docs/platform/mssql/installation.mdx

```diff
@@ -158,6 +159,15 @@ installation.
     PROVIDER_KEY_NAME = 'ekm-encryption-key';
     ```
 
+    !> **Warning:** Encountering the SQL error response `Provider error code: 2050`
+    at this stage is typically caused by an incorrect license that's lacking
+    "Key Management Transparent Data Encryption". Using the Vault Audit Logs confirm
+    that EKM is in network reach to Vault API path `/sys/license/status`. Failing 
+    attempts by EKM to connect to Vault in order to check its license or the use 
+    of the wrong license are likely causes for the `2050` error response. This 
+    would need to be recitified for the above SQL to be successful before 
+    commencing any farther.
```

Comment:
I don't think a danger alert is called for here. I'd also prefer to refer to the troubleshooting page so we can go into more detail there on the possible causes without interrupting the flow too much.

```suggestion
    -> **Note:** This is the first step at which the EKM provider will communicate with Vault. If
    Vault is misconfigured, this step is likely to fail. See
    [troubleshooting](/docs/platform/mssql/troubleshooting) for tips on specific error codes.
```

For the troubleshooting page, I think we should add a new section titled "Error Codes", and just put 2050 as the only entry for now, but I'll add more error codes at a later date.

```
## Error Codes

During installation, the EKM provider registers a manifest of coded event logs to aid debugging. You may see the following error codes during operation.

### 2050 License Error

The EKM provider was unable to verify that Vault has the correct license features. This
could be due to:

- An incompatible Vault Enterprise license - see the installation [prerequisites](/docs/platform/mssql/installation#prerequisites) for the required license feature.
- Lack of network connectivity - Check Vault's audit logs to see if any requests are made to
   authenticate via AppRole or query the `/sys/license/status` API.
- Misconfigured AppRole auth - Ensure you provided the correct Role ID and Secret ID when
   configuring the SQL Server `CREDENTIAL`. See the
   [installation instructions](/docs/platform/mssql/installation) for an end-to-end working example.
- Incorrect policy permissions - The EKM provider requires the `read` capability
   on the path `sys/license/status`. See the `tde-policy` created in the
   [installation instructions](/docs/platform/mssql/installation#configuring-vault)
   for an example of a working policy.
```

Reply:
Please could we add the Error Codes section to the trouble shooting page as well? You're welcome to copy my suggestion wholesale or make edits as you like, but I just want to make sure we're linking to a page with something helpful on it for this specific callout.

Reply:
Hey @tomhjp  I've added your text verbatim - sorry for the delay.
<img width="1252" alt="Screenshot 2022-07-01 at 09 35 42" src="https://user-images.githubusercontent.com/974854/176847688-1ebcb372-f9f5-48cb-88eb-a612fa234fbb.png">

Reply:
Hey @tomhjp I've added your text verbatim - sorry for the delay.
<img width="1056" alt="Screenshot 2022-07-01 at 09 40 51" src="https://user-images.githubusercontent.com/974854/176848490-10ebf666-2700-4409-b7db-536acdc154e6.png">